pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -712,10 +712,13 @@ func (n *NodeInfo) FilterOutPods(pods []*v1.Pod) []*v1.Pod {
 			continue
 		}
 		// If pod is on the given node, add it to 'filtered' only if it is present in nodeInfo.
-		podKey, _ := GetPodKey(p)
+		var podKey string
+		var err error
+		if podKey, err = GetPodKey(p); err != nil {
+			continue
+		}
 		for _, np := range n.Pods() {
-			npodkey, _ := GetPodKey(np)
-			if npodkey == podKey {
+			if npodkey, err := GetPodKey(np); err == nil && npodkey == podKey {
```

Comment:
Earlier I submitted a PR which used map for lookup which was turned down by @Huang-Wei 

I can raise another PR if map is deemed better in this case - since this PR is about checking error return.